src/nodes.coffee

```diff
@@ -4573,6 +4573,53 @@ exports.Switch = class Switch extends Base
     fragments.push @makeCode @tab + '}'
     fragments
 
+  astType: -> 'SwitchStatement'
+
+  casesAst: (o) ->
```

Comment:
Should the cases perhaps be their own node class?

Reply:
So that's the thing I think would be tricky

This PR introduces a new `SwitchCase` node class, which basically corresponds to the node class we'd want here in order to be able to do a more typical child-node AST generation

The issue is that it's not a one-to-one mapping from our `SwitchCase` node class objects to AST `SwitchCase` nodes. So in order to try and set it up for typical child-node AST generation, we'd need to transform our `SwitchCase` node class objects into something, I guess:
1. They could map to different instances of our same `SwitchCase` node class, but that seems a little weird/confusing perhaps?
1. Or we could introduce a separate eg `SwitchCaseAst` node class, but that also seems kind of weird/overkill?

Additionally, whichever node class we were mapping to, we'd also need to map our `@otherwise` into one of them since it needs to generate a `SwitchCase` AST node (that's how the `default:` case is represented in Babel-style ASTs)

And either way, it seems possible that the mapping might be even messier than the AST generation logic below (with its location data wrangling)

But I could try one of these approaches and we could compare, we certainly do usually end up feeling like it's cleaner to avoid manually assembling AST nodes whenever possible